CMakeLists.txt

```diff
@@ -13,52 +14,74 @@ file(GLOB semseterser_src
 )
 
 # Executable name to be built
-add_executable(semseterser ${semseterser_src})
+add_executable(${PROJECT_NAME} ${semseterser_src})
 
 # Compiler options per platform
 if(MSVC)
-    target_compile_options(semseterser PRIVATE /W4 -O2)
+    target_compile_options(${PROJECT_NAME} PRIVATE /W4 -O2)
 else()
-    target_compile_options(semseterser PRIVATE -Wall -Wextra -Wpedantic -O2)
+    target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra -Wpedantic -O2)
 endif()
 
+include(GNUInstallDirs)
+
+# Enable downloading external projects during build
+include(ExternalProject)
+
 # Enable running ctest
 enable_testing()
 
 # Add googletest. Will be downloaded from git.
-include(ExternalProject)
 ExternalProject_Add(
     googletest
     GIT_REPOSITORY "https://github.com/google/googletest.git"
     GIT_TAG "master"
     CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/googletest
 )
 
-# Exclude googletest building from target "all"
-set_target_properties(googletest PROPERTIES EXCLUDE_FROM_ALL TRUE)
-
-include(GNUInstallDirs)
+# Required for googletest
+find_package(Threads REQUIRED)
 
 link_directories(${CMAKE_BINARY_DIR}/googletest/${CMAKE_INSTALL_LIBDIR})
 include_directories(${CMAKE_BINARY_DIR}/googletest/${CMAKE_INSTALL_INCLUDEDIR})
 
-# Required for googletest
-find_package(Threads REQUIRED)
-
 # Use all .cpp files in the test folder
 file(GLOB_RECURSE TEST_SOURCES
     RELATIVE ${PROJECT_SOURCE_DIR}
     CONFIGURE_DEPENDS
     src/test/*.cpp)
 
-set(PROJECT_TESTER_NAME ${PROJECT_NAME}_test)
 add_executable(${PROJECT_TESTER_NAME} ${TEST_SOURCES})
 add_dependencies(${PROJECT_TESTER_NAME} googletest)
-target_link_libraries(${PROJECT_TESTER_NAME} gtest gtest_main Threads::Threads)
+target_link_libraries(
+    ${PROJECT_TESTER_NAME}
+    gtest
+    gtest_main
+    Threads::Threads
+)
 
 add_test(${PROJECT_TESTER_NAME} ${PROJECT_TESTER_NAME})
 
 # Add make tester that builds tests and runs them
 add_custom_target(tester
     COMMAND ${PROJECT_TESTER_NAME} || true # suppress verbose errors
     DEPENDS ${PROJECT_TESTER_NAME})
+
+# Exclude googletest building from target "all"
+set_target_properties(googletest PROPERTIES EXCLUDE_FROM_ALL TRUE)
+
+# Add SFML
+ExternalProject_Add(
+    sfml
+    GIT_REPOSITORY "https://github.com/SFML/SFML.git"
+    GIT_TAG "2.5.1"
+    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_BINARY_DIR}/sfml
+)
+
+find_package(SFML 2.5.1 COMPONENTS graphics REQUIRED)
```

Comment:
This line caused the build to fail on my machine locally as well as in Travis:
```
CMake Error at CMakeLists.txt:81 (find_package):

  By not providing "FindSFML.cmake" in CMAKE_MODULE_PATH this project has

  asked CMake to find a package configuration file provided by "SFML", but

  CMake did not find one.

  Could not find a package configuration file provided by "SFML" (requested

  version 2.5.1) with any of the following names:

    SFMLConfig.cmake

    sfml-config.cmake

  Add the installation prefix of "SFML" to CMAKE_PREFIX_PATH or set

  "SFML_DIR" to a directory containing one of the above files.  If "SFML"

  provides a separate development package or SDK, be sure it has been

  installed.
```

After I removed it, the build finished OK.

Also, you should add the installation of the dependencies required by SFML to `.travis.yml`, I bet it will need 'em as well.

Reply:
Ah, I had SFML installed locally, that's why it worked on my machine. I shall look into it.